File: 01/solution2.jl
function main(args)

    f = open(args[1]);
    input = parse.(Int64,readlines(f))
    close(f)

    n = length(input)
    M = zeros((n,n,n))
    for i = 1:n
        for j = 1:n
            for k = 1:n
                M[i,j,k] = input[i] + input[j] + input[k]
            end
        end
    end

    idx = findall(x->x==2020,M)[1]
    solution = input[idx[1]] * input[idx[2]] * input[idx[3]]
    println(solution)
end

main(ARGS)

    

File: 01/solution2.py
import numpy as np
import time

t = time.time()
input = np.loadtxt("./input.txt", dtype=np.int64)

length = np.size(input)

M = np.zeros((length, length, length))
for i in range(length):
    for j in range(length):
        for k in range(length):
            M[i,j,k] = input[i] + input[j] + input[k]

idx = np.where(M==2020)
solution = input[idx[0][0]] * input[idx[1][0]] * input[idx[2][0]]
elapsed = time.time() - t
print(solution)
print("elapsed time: ", elapsed)